Model/MenuRepository.php

```diff
@@ -1,66 +1,117 @@
 <?php
 namespace Snowdog\Menu\Model;
 
-use Snowdog\Menu\Api\MenuRepositoryInterface;
-use Snowdog\Menu\Api\Data\MenuInterface;
-use Snowdog\Menu\Model\ResourceModel\Menu\CollectionFactory;
-use Snowdog\Menu\Api\Data\MenuSearchResultsInterfaceFactory;
-
 use Magento\Framework\Api\SearchCriteriaInterface;
+use Magento\Framework\Api\SortOrder;
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Exception\CouldNotDeleteException;
 use Magento\Framework\Exception\CouldNotSaveException;
 use Magento\Framework\Exception\NoSuchEntityException;
-use Magento\Framework\Exception\CouldNotDeleteException;
+use Snowdog\Menu\Api\Data\MenuInterface;
+use Snowdog\Menu\Api\Data\MenuSearchResultsInterfaceFactory;
+use Snowdog\Menu\Api\Data\NodeInterface;
+use Snowdog\Menu\Api\MenuRepositoryInterface;
+use Snowdog\Menu\Model\ResourceModel\Menu\CollectionFactory as MenuCollectionFactory;
+use Snowdog\Menu\Model\ResourceModel\Menu\Node\Collection as NodeCollection;
+use Snowdog\Menu\Model\ResourceModel\Menu\Node\CollectionFactory as NodeCollectionFactory;
 
 class MenuRepository implements MenuRepositoryInterface
 {
-    protected $objectFactory;
+    /** @var MenuFactory */
+    protected $menuFactory;
+
+    /** @var CollectionFactory */
     protected $collectionFactory;
 
-    /**
-     * @var MenuSearchResultsInterfaceFactory
-     */
+    /** @var MenuSearchResultsInterfaceFactory */
     private $menuSearchResultsFactory;
 
+    /** @var ResourceModel\Menu */
+    private $menuResourceModel;
+
+    /** @var NodeCollectionFactory|null */
+    private $nodeCollectionFactory;
+
+    /** @var ResourceModel\Menu\Node|null */
+    private $nodeResourceModel;
+
+    /**
+     * @param MenuFactory $menuFactory
+     * @param MenuCollectionFactory $menuCollectionFactory
+     * @param MenuSearchResultsInterfaceFactory $menuSearchResults
+     * @param ResourceModel\Menu|null $menuResourceModel
+     * @param NodeCollectionFactory|null $nodeCollectionFactory
+     */
     public function __construct(
-        MenuFactory $objectFactory,
-        CollectionFactory $collectionFactory,
-        MenuSearchResultsInterfaceFactory $menuSearchResults
+        MenuFactory $menuFactory,
+        MenuCollectionFactory $menuCollectionFactory,
+        MenuSearchResultsInterfaceFactory $menuSearchResults,
+        ResourceModel\Menu $menuResourceModel = null,
+        NodeCollectionFactory $nodeCollectionFactory = null,
+        ResourceModel\Menu\Node $nodeResourceModel = null
```

Comment:
PHPDoc is missing for `$nodeResourceModel`

Reply:
✔️  Sorted